Sources/MongoSwift/BSON/Document.swift

```diff
@@ -395,8 +422,12 @@ extension Document {
                 if let newValue = newValue {
                     try self.setValue(for: key, to: newValue)
                 } else {
-                    // TODO: SWIFT-224: use va_list variant of bson_copy_to_excluding to improve performance
-                    self = self.filter { $0.key != key }
+                    let bsonOpt = bson_new()
+                    withVaList([]) { bson_copy_to_excluding_noinit_va(self._bson, bsonOpt, key, $0) }
+                    guard let bson = bsonOpt else {
```

Comment:
I think it would probably be better to check that the return value of `bson_new` is non-null before passing it into `bson_copy_to_excluding_noinit_va`. if it is null, we would likely crash in that method before getting to the fatal error.

that said, we will probably never even hit the earlier fatal error, either.

`bson_new()` returning an optional value here is sort of funny - Swift translates the signature this way because there's nothing stopping you from just returning `NULL` in place of a `bson_t *` in C, but in practice, the method will never actually do so. It may fail to allocate memory, but in that case libmongoc will send SIGABRT anyway and the method won't return: https://github.com/mongodb/mongo-swift-driver/blob/33aed8834adb1c07d811b3431ceceee1355784a4/Sources/CLibMongoC/bson/bson-memory.c#L72

so, even if we move the error up we probably still won't ever encounter it, but I think it's a bit more future-proof in case `bson_new` ever did start returning NULL without a SIGABRT being sent first.

(same applies to a couple other places you do this)

Reply:
Done.